### PR TITLE
Implemented the web project for Assignment 1

### PR DESCRIPTION
**Pull request by TheXGeneral**
Created a web page that handles the functionality of seeing menu items of a restaurant
Styled the page according to the given design
Implemented functionalities for the page, both for loading items into the page and for performing CRUD operations for the items (aka editing list of items dashboard)

Link for github pages: https://thexgeneral.github.io/Wolpack_A1/

### CODE REVIEW

File: css/rSlider.min.css [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+.rs-container .rs-pointer::after,
+.rs-container .rs-pointer::before {
+  content: "";
+  position: absolute;
+  width: 1px;
+  height: 9px;
+  background-color: #ddd;
+  left: 12px;
+  top: 5px;
+}
```
Comment (AdrianFlorian) [inline css/rSlider.min.css:63]: please keep the spacing and new lines consistent :-)

File: js/data.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+//import Item from "./domain.js";
+function uuidv4() {
+    return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+      (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
+    );
+  }
+let Data=[
+    {id:uuidv4(),photo_link:"img/food.jpg",name:"Pasta Carbonara",category:"Second courses",price:2},
```
Comment (AdrianFlorian) [inline js/data.js:8]: spacing :-), space after `,` please

File: js/edit.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+//CRUD
+function add_new_element(){
+    let _id=uuidv4();
+    let _element={id:_id,photo_link:"",name:"",category:"",price:0};
+    edit_add_item_into_table({id:_id,photo_link:"",name:"",category:"",price:0});
+    let Data=JSON.parse(localStorage.getItem("list_of_elements"));
+    Data.push(_element);
+    localStorage.setItem("list_of_elements", JSON.stringify(Data));
+    edit_item(_id);
+}
```
Comment (AdrianFlorian) [inline js/edit.js:25]: same here new line :-)

File: js/edit.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
+function open_edit_menu(){
```
Comment (AdrianFlorian) [inline js/edit.js:1]: function names in JS have camel case `openEditMenu`

File: js/edit.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+}
+function update_item(id,new_photo_link,new_name,new_category,new_price){
+    let Data=JSON.parse(localStorage.getItem("list_of_elements"));
+    let position_to_update=0;
+    while(Data[position_to_update].id!=id)
+        position_to_update++;
+    Data[position_to_update]["photo_link"]=new_photo_link;
+    Data[position_to_update]["name"]=new_name;
+    Data[position_to_update]["category"]=new_category;
+    Data[position_to_update]["price"]=new_price;
```
Comment (AdrianFlorian) [inline js/edit.js:47]: why not put this in a new variable? `Data[position_to_update]` this way you will not have to duplicate the code :-)

**Review by AdrianFlorian — CHANGES_REQUESTED** [review]
Nice job 👏

Overall comments: 
- watch out for the spacing
- please rename the PR (pull request) name to reflect the functionality + update the description of the PR
- naming conventions are important
- avoid duplicated code as much as possible :-)

File: js/edit.js (hunk 4 above)
Comment (TheXGeneral) [reply js/edit.js]: Now, all functions in Js use camelCase and all classes in CSS use kebab-case

File: js/data.js (hunk 2 above)
Comment (TheXGeneral) [reply js/data.js]: I added space after every comma and before and after every =, ==, etc.

File: js/data.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+//import Item from "./domain.js";
```
Comment (AdrianFlorian) [inline js/data.js:1]: why the comment? :-)

File: js/edit.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+function updateTableButtons(id,_text,_function){
+    document.getElementById(`edit_save_button${id}`).innerHTML = _text;
+    document.getElementById(`edit_save_button${id}`).addEventListener("click", function() {
+        _function(id);
+    }, false);
+}
+
+//CRUD
```
Comment (AdrianFlorian) [inline js/edit.js:18]: no need to have unnecessary comments

File: js/edit.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    for(let i = 1; i < columns.length-1; i++){
+        columns[i].contentEditable = "false";
+        columns[i].style.border = "0px";
+        content.push(columns[i].innerHTML);
+    }
+    updateItem(...content);
+    updateTableButtons(id, "Edit", editItem);
+}
+
+//loading data into the menu
```
Comment (AdrianFlorian) [inline js/edit.js:69]: If you feel the need to add a comment above a method name it means that it's name is not descriptive enough. You 
If you ever need to add comments inside a method, it means it's trying to do too much and it needs to be broken down into multiple smaller methods with descriptive names. 🤘

File: js/index.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+function loadItems(){
+    let where_to_append = document.getElementById("items");
+    where_to_append.innerHTML = "";
+    let max_price = 0;
+    let min_price = 10000;
+    let _list_of_categories = new Set();
+    [max_price, min_price, _list_of_categories] = loadItemsWithoutUpdatingCommands(JSON.parse(localStorage.getItem("list_of_elements")));
+    //TODO implement set raange limits function
```
Comment (AdrianFlorian) [inline js/index.js:77]: remove comment

**Review by AdrianFlorian — CHANGES_REQUESTED** [review]
Nice job Dan, almost there!

Please watch out for the comments, a good code should never need comments.

Also, please add a descriptive title to the PR and a good description. Check https://github.com/cristichirtos/RoR-internship-assignment-1/pull/1 as an inspiration :-)

**Review by chioreandan — CHANGES_REQUESTED** [review]
I forgot to submit the review. Some comments might not be applicable

File: js/edit.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    editAddItemIntoTable({id:_id, photo_link:"", name:"", category:"", price:0});
+    let Data = JSON.parse(localStorage.getItem("list_of_elements"));
+    Data.push(_element);
+    localStorage.setItem("list_of_elements", JSON.stringify(Data));
+    editItem(_id);
+}
+
+function editItem(id){
+    let columns = document.getElementById(id).children;
+    for(let i = 1; i < columns.length-1; i++){
```
Comment (chioreandan) [inline js/edit.js:31]: use ``` forEach() ```

File: index.html [hunk 11]
```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
```
Comment (chioreandan) [inline index.html:1]: Change your tab spacing to 2 instead of 4

File: css/index.css [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+  --second-accent-color: #444444;
+  --third-accent-color: #888888;
+}
+
+* {
+  padding: 0px;
+  margin: 0px;
+}
+
+/*scrollbar*/
```
Comment (AdrianFlorian) [inline css/index.css:15]: do you need this comment? :-)

**Review by AdrianFlorian — APPROVED** [review]
@TheXGeneral nice job!  👏

**Review by chioreandan — APPROVED** [review]
Nice job @TheXGeneral 👍. Please also check the conflicts that you have.

**Pull request merged**